packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -90,7 +90,20 @@ export default class Proxy {
   async destroyProxy(id) {
     const { vmUuid } = await this._getProxy(id)
     if (vmUuid !== undefined) {
-      await this._app.getXapi(vmUuid).deleteVm(vmUuid)
+      try {
+        await this._app.getXapi(vmUuid).deleteVm(vmUuid)
+      } catch (error) {
+        if (!noSuchObject.is(error)) {
+          throw error
+        }
+      }
+
+      this._app
+        .unbindLicense({
+          boundObjectId: vmUuid,
+          productId: this._xoProxyConf.licenseProductId,
+        })
+        .catch(log.warn)
```

Comment:
This PR contains two changes, either:
- split into two PRs
- split into two commits and explain that this PR must not be squashed
- edit the commit message to clarify that it contains two changes